#This patch file is from Qualcomm Innovation Center, Inc. and is provided under the following license:
#
#Copyright (c) 2025 Qualcomm Innovation Center, Inc. All rights reserved.
#SPDX-License-Identifier: BSD-3-Clause-Clear
From 4dcc4e1968c547282b9c3a577a008d01f7dfc7b0 Mon Sep 17 00:00:00 2001
From: weijshen <quic_weijshen@quicinc.com>
Date: Fri, 25 Apr 2025 17:34:17 +0800
Subject: [PATCH] {jazzy}: remove outdated appends

1. Fix ros bags convert runtime issue.
Log:
Error string: Could not load library dlopen error: /usr/lib/
librosbag2_compression_zstd.so: undefined symbol: ZSTD_getFrameContentSize.
2. Fix osqp vendor fetch issue.
Log:
Unable to find revision dd3878565d2be8dcd2b23e76a08e95b1d2baf89f in branch
3. Fix google-benchmark-vendor do patch issue.
Log:
Patch does not apply

Upstream-Status: Pending

Signed-off-by: weijshen <quic_weijshen@quicinc.com>
---
 .../google-benchmark-vendor_0.5.0-2.bbappend     |  4 ----
 .../osqp-vendor/osqp-vendor_0.2.0-4.bbappend     | 16 ----------------
 .../rosbag2-compression-zstd_0.26.5-1.bbappend   |  9 ---------
 .../rosbag2/zstd-vendor_0.26.5-1.bbappend        |  9 ---------
 4 files changed, 38 deletions(-)
 delete mode 100644 meta-ros2-jazzy/recipes-bbappends/google-benchmark-vendor/google-benchmark-vendor_0.5.0-2.bbappend
 delete mode 100644 meta-ros2-jazzy/recipes-bbappends/osqp-vendor/osqp-vendor_0.2.0-4.bbappend
 delete mode 100644 meta-ros2-jazzy/recipes-bbappends/rosbag2/rosbag2-compression-zstd_0.26.5-1.bbappend
 delete mode 100644 meta-ros2-jazzy/recipes-bbappends/rosbag2/zstd-vendor_0.26.5-1.bbappend

diff --git a/meta-ros2-jazzy/recipes-bbappends/google-benchmark-vendor/google-benchmark-vendor_0.5.0-2.bbappend b/meta-ros2-jazzy/recipes-bbappends/google-benchmark-vendor/google-benchmark-vendor_0.5.0-2.bbappend
deleted file mode 100644
index 2077c4a29..000000000
--- a/meta-ros2-jazzy/recipes-bbappends/google-benchmark-vendor/google-benchmark-vendor_0.5.0-2.bbappend
+++ /dev/null
@@ -1,4 +0,0 @@
-# Copyright (c) 2020 LG Electronics, Inc.
-
-FILESEXTRAPATHS:prepend := "${THISDIR}/${BPN}:"
-SRC_URI += "file://0001-CMakeLists.txt-prevent-building-google-benchmark-wit.patch"
diff --git a/meta-ros2-jazzy/recipes-bbappends/osqp-vendor/osqp-vendor_0.2.0-4.bbappend b/meta-ros2-jazzy/recipes-bbappends/osqp-vendor/osqp-vendor_0.2.0-4.bbappend
deleted file mode 100644
index fb2f7317b..000000000
--- a/meta-ros2-jazzy/recipes-bbappends/osqp-vendor/osqp-vendor_0.2.0-4.bbappend
+++ /dev/null
@@ -1,16 +0,0 @@
-# Copyright (c) 2020 LG Electronics, Inc.
-
-# Instead of fetching
-# https://github.com/oxfordcontrol/osqp.git from tag v0.6.2
-# during do_compile
-FILESEXTRAPATHS:prepend := "${THISDIR}/${BPN}:"
-SRC_URI = " \
-    git://github.com/ros2-gbp/osqp_vendor-release;name=release;${ROS_BRANCH};protocol=https \
-    file://0002-CMakeLists.txt-fetch-osqp-with-bitbake-fetcher.patch \
-    git://github.com/oxfordcontrol/osqp.git;protocol=https;name=osqp;destsuffix=git/osqp-upstream;branch=master \
-    git://github.com/oxfordcontrol/qdldl.git;protocol=https;name=qdldl;destsuffix=git/osqp-upstream/lin_sys/direct/qdldl/qdldl_sources;branch=master \
-"
-SRCREV_release = "dd3878565d2be8dcd2b23e76a08e95b1d2baf89f"
-SRCREV_osqp = "f9fc23d3436e4b17dd2cb95f70cfa1f37d122c24"
-SRCREV_qdldl = "7d16b70a10a152682204d745d814b6eb63dc5cd2"
-SRCREV_FORMAT = "release_osqp_qdldl"
diff --git a/meta-ros2-jazzy/recipes-bbappends/rosbag2/rosbag2-compression-zstd_0.26.5-1.bbappend b/meta-ros2-jazzy/recipes-bbappends/rosbag2/rosbag2-compression-zstd_0.26.5-1.bbappend
deleted file mode 100644
index 60f28ad9f..000000000
--- a/meta-ros2-jazzy/recipes-bbappends/rosbag2/rosbag2-compression-zstd_0.26.5-1.bbappend
+++ /dev/null
@@ -1,9 +0,0 @@
-# Copyright (c) 2020 LG Electronics, Inc.
-
-FILESEXTRAPATHS:prepend := "${THISDIR}/${BPN}:"
-SRC_URI += "file://0001-CMakeLists.txt-drop-dependency-on-zstd_vendor.patch"
-
-# PN package in zstd-vendor is empty and not created, remove runtime dependency on it
-ROS_EXEC_DEPENDS:remove = "zstd-vendor"
-
-inherit pkgconfig
diff --git a/meta-ros2-jazzy/recipes-bbappends/rosbag2/zstd-vendor_0.26.5-1.bbappend b/meta-ros2-jazzy/recipes-bbappends/rosbag2/zstd-vendor_0.26.5-1.bbappend
deleted file mode 100644
index 14dbaf96c..000000000
--- a/meta-ros2-jazzy/recipes-bbappends/rosbag2/zstd-vendor_0.26.5-1.bbappend
+++ /dev/null
@@ -1,9 +0,0 @@
-# Copyright (c) 2020 LG Electronics, Inc.
-# Copyright (c) 2023 Wind River Systems, Inc.
-
-FILESEXTRAPATHS:prepend := "${THISDIR}/${BPN}:"
-SRC_URI += "file://0001-CMakeLists.txt-prevent-building-zstd-with-ExternalPr.patch"
-
-inherit pkgconfig
-
-LICENSE = "Apache-2.0"
--
2.34.1

